d2compiler/compile.go

```diff
@@ -329,7 +336,7 @@ func (c *compiler) compileField(obj *d2graph.Object, f *d2ir.Field) {
 			Scope:           fr.Context.Scope,
 			ScopeAST:        fr.Context.ScopeAST,
 		}
-		if fr.Context.ScopeMap != nil {
+		if fr.Context.ScopeMap != nil && !d2ir.IsVar(fr.Context.ScopeMap) {
```

Comment:
Instead, strip the IR of `vars` as you add them to the varStack. That way the compiler won't see vars.

Reply:
when i do spread substitution, i'm overlaying the var map. that retains all the parent scope, field references, etc. it seems annoying to clean it.

Reply:
i also don't think i should be modifying the IR like that. like for language tooling, going to def of a var requires it be present in the IR

Reply:
Fair enough.